packages/sdk/src/dataProtector/sharing/setSubscriptionParams.ts

```diff
@@ -32,3 +55,27 @@ export const setSubscriptionParams = async ({
     );
   }
 };
+
+async function checkAndGetCollection({
+  graphQLClient,
+  collectionTokenId,
+  userAddress,
+}: {
+  graphQLClient: GraphQLClient;
+  collectionTokenId: number;
+  userAddress: Address;
+}) {
+  const collection = await getCollectionById({
+    graphQLClient,
+    collectionTokenId: toHex(collectionTokenId),
+  });
+
```

Comment:
Add a check, if the collection simply does not exist?

Reply:
Commit: b59fb35e00bca232bab042e498f47cca3c0f137e